From 7b97a34ed7281f433606196cb9b2e37492360df0 Mon Sep 17 00:00:00 2001
From: LoveSy <shana@zju.edu.cn>
Date: Fri, 12 Apr 2024 23:46:38 +0800
Subject: [PATCH 1/2] Prevent serial of ro props from self increment

This avoids triggering update of ro props for apps using property_wait,
and matches original behavior.

Co-authored-by: canyie <a1364259@163.com>
Change-Id: I05b5a2b1da9f895fdc1b2f7f15942dbfcc0cde05
Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
---
 libc/system_properties/system_properties.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libc/system_properties/system_properties.cpp b/libc/system_properties/system_properties.cpp
index 6924ff145..d5c6b6fd1 100644
--- a/libc/system_properties/system_properties.cpp
+++ b/libc/system_properties/system_properties.cpp
@@ -318,7 +318,7 @@ int SystemProperties::Update(prop_info* pi, const char* value, unsigned int len)
   }
   // Now the primary value property area is up-to-date. Let readers know that they should
   // look at the property value instead of the backup area.
-  int new_serial = (len << 24) | ((serial + 1) & 0xffffff);
+  int new_serial = (len << 24) | ((is_read_only(pi->name) ? 0u : (serial + 1)) & 0xffffff);
   atomic_store_explicit(&pi->serial, new_serial, memory_order_release);
   if (have_override) {
     atomic_store_explicit(&override_pi->serial, new_serial, memory_order_relaxed);
-- 
2.52.0

